geopyspark/avroserializer.py

```diff
@@ -10,13 +9,10 @@
 class AvroSerializer(FramedSerializer):
     def __init__(self,
             schema_json,
-            geotrellis_decoder=GeoTrellisDecoder(),
-            geotrellis_encoder=GeoTrellisEncoder()):
+            avroregistry=AvroRegistry()):
```

Comment:
This is a gotcha. The registry is going to be created once during method definition and re-used in all constructors. Any state change to this object will be available to all future instance of `AvroSerializer`. The pattern around it seems to be to set the default value to `None` and check for that in the body. (ref: http://docs.python-guide.org/en/latest/writing/gotchas/)

But `AvroRegistry` should have a `DEFAULT` fields to avoid multiple re-instantiations which may be expensive in this case.